### PR TITLE
DOP-2317: add support for hidden flag to readconcern and method rstobjects

### PR DESCRIPTION
**Pull request by schmalliso**

### CODE REVIEW

File: snooty/specparser.py [hunk 1]
```diff
@@ -109,7 +62,7 @@ class RefRoleType:
 
 #: Spec definition of a role: this can be either a PrimitiveRoleType, or
 #: an object requiring additional configuration.
-RoleType = Union[PrimitiveRoleType, LinkRoleType, RefRoleType]
+RoleType = Union["PrimitiveRoleType", "LinkRoleType", "RefRoleType"]
```
Comment (i80and) [inline snooty/specparser.py:65]: I suspect without proof this might be causing the test failure

Comment (schmalliso) [reply snooty/specparser.py:65]: It's only failing on 3.8, which is why I didn't catch it when I ran `make test` locally, which is INTERESTING…

Comment (i80and) [reply snooty/specparser.py:65]: what happens if you remove the double quotes? In *theory* our `from __future__ import annotations` incantation at the top of the file should make that work

**Review by i80and — APPROVED** [review]
:shipit:

**Pull request merged**